### PR TITLE
fix(hooks): price Sonnet 5 at the published $2/$10 rate

### LINKED ISSUE #2724: cost-tracker: Sonnet rate is stale for Sonnet 5's introductory pricing (expires 2026-08-31)
`RATE_TABLE.sonnet` in `scripts/hooks/cost-tracker.js` is `{ in: 3.00, out: 15.0, cacheWrite: 3.75, cacheRead: 0.30 }`. That is the post-2026-09-01 standard Sonnet rate. Claude Sonnet 5 is currently on introductory pricing, $2/$10 per million input/output tokens ($2.50 5m cache write, $0.20 cache read), through August 31, 2026: https://platform.claude.com/docs/en/docs/about-claude/pricing#claude-sonnet-5-introductory-pricing. `getRates()` has one `sonnet` bucket for every Sonnet model, so any Sonnet 5 session priced today overcounts by 50%.

Repro (Docker, `node:20-slim`, table and function copied verbatim from HEAD `59a99d66`):

```js
const rates = getRates('claude-sonnet-5-20260514'); // { in: 3, out: 15, cacheWrite: 3.75, cacheRead: 0.3 }
// 1M in / 1M out -> $18.00 reported, $12.00 actual at the published introductory rate (50% over)
```

Same root cause as #2574, a hard-coded table with no per-model-version disambiguation, different symptom: #2574 is Opus stale by 3x and Fable falling through to Sonnet; this is Sonnet 5 priced at its own post-window rate three weeks early. Neither my fix in #2596 nor the open #2621/#2691/#2692 touch it. #2691 actually pins the current $3/$15 as correct with a new test ("prices Sonnet at $3/$15 and marks it as a real match"), which is right starting 2026-09-01 but wrong until then.

The fix needs a calendar check the other rows don't (this is a fixed-window promo, not a version distinction), and #2691 is about to restructure this same table, so I'm filing separately rather than folding it into #2596. Happy to send a follow-up PR once #2596/#2691 settle, or sooner if you'd rather have it now.

### PR DESCRIPTION
**Pull request by santhiprakash**
## What Changed

`scripts/hooks/cost-tracker.js` now prices `claude-sonnet-5` (and its dated variants) at the published **$2 / $10 per 1M tokens** rate, instead of falling through to the older Sonnet bucket at **$3 / $15**.

- Added a `sonnet5` row to `RATE_TABLE`: `{ in: 2.00, out: 10.0, cacheWrite: 2.50, cacheRead: 0.20 }`.
- Added `isSonnet5()` with a word-boundary regex so `claude-sonnet-5-20260514` matches but `claude-sonnet-4-6` does not.
- Added two regression tests:
  - `claude-sonnet-5` 1M in / 1M out costs $12.00
  - `claude-sonnet-4-6` 1M in / 1M out still costs $18.00

## Why This Change

Claude Sonnet 5 is billed at $2 input / $10 output per million tokens. Anthropic's pricing page now lists this as the standard rate and states that the previously scheduled September 1, 2026 increase to $3/$15 will not occur (https://platform.claude.com/docs/en/about-claude/pricing#claude-sonnet-5-introductory-pricing). The existing `RATE_TABLE` had no Sonnet 5 row, so `getRates()` returned the legacy `sonnet` ($3/$15) rate for any Sonnet 5 session, overcounting spend by 50%.

This touches the canonical `cost-tracker.js` rate source, not the unused `cost-estimate.js` duplicate noted in #2656 / #2777.

## Testing Done

- [x] Targeted automated tests pass locally:
  ```bash
  node tests/hooks/cost-tracker.test.js
  ```
  Result: `Results: Passed: 12, Failed: 0`
- [x] ESLint passes on changed files.
- [x] Edge cases considered and tested: Sonnet 5 word-boundary matching, Sonnet 4.6 unaffected.

## Type of Change

- [x] `fix:` Bug fix
- [ ] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [ ] `docs:` Documentation
- [x] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist

- [x] No secrets or API keys committed
- [x] Follows conventional commits format
- [x] Pre-commit checks pass (eslint on changed files)

Fixes #2724

### CODE REVIEW

**Comment by santhiprakash** [comment]
The calendar cutoff is no longer needed. Anthropic’s live pricing page now states that the $2/$10 Sonnet 5 rate, announced as introductory through August 31, 2026, is the standard price, and the previously scheduled September 1 increase to $3/$15 will not occur:

> "The $2/$10 per million input/output token pricing for Claude Sonnet 5, announced at launch as introductory pricing through August 31, 2026, is now the standard price. The previously scheduled increase to $3/$15 per million input/output tokens on September 1, 2026 will not occur."

Source: https://platform.claude.com/docs/en/about-claude/pricing#claude-sonnet-5-introductory-pricing

The PR description already cites this. No date-aware logic is required.

File: scripts/hooks/cost-tracker.js [hunk 1]
```diff
@@ -71,15 +71,21 @@ function readHarnessCost(sessionId, maxAgeSeconds) {
...
+function isSonnet5(model) {
+  return /(?:^|[^a-z0-9])sonnet-5(?:[^a-z0-9]|$)/.test(model);
+}
+
 function getRates(model) {
   const m = String(model || '').toLowerCase();
-  if (m.includes('haiku')) return RATE_TABLE.haiku;
-  if (m.includes('opus'))  return RATE_TABLE.opus;
+  if (m.includes('haiku'))  return RATE_TABLE.haiku;
+  if (isSonnet5(m))        return RATE_TABLE.sonnet5;
```
[the remark this answers is not in the document]
Comment (santhiprakash) [reply scripts/hooks/cost-tracker.js:87]: No date condition is needed. Anthropic’s pricing page now states the $2/$10 Sonnet 5 rate, announced as introductory through August 31, 2026, is the standard price and the scheduled September 1 increase to $3/$15 will not occur. Source: https://platform.claude.com/docs/en/about-claude/pricing#claude-sonnet-5-introductory-pricing

File: scripts/hooks/cost-tracker.js [hunk 2]
```diff
@@ -71,15 +71,21 @@ function readHarnessCost(sessionId, maxAgeSeconds) {
...
+  if (m.includes('opus'))   return RATE_TABLE.opus;
```
[the remark this answers is not in the document]
Comment (santhiprakash) [reply scripts/hooks/cost-tracker.js:88]: scripts/lib/cost-estimate.js has no live callers and is intentionally not being updated. Issue #2656 and open PR #2777 (remove unused cost-estimate.js duplicate rate table) track its removal. This PR follows the issue requirement to update only the canonical cost-tracker rate source.

File: tests/hooks/cost-tracker.test.js [hunk 3]
```diff
@@ -297,7 +297,63 @@ function runTests() {
...
+      withTempHome(tmpHome)
+    );
+    assert.strictEqual(result.code, 0, `Expected exit code 0, got ${result.code}`);
+
+    const metricsFile = path.join(tmpHome, '.claude', 'metrics', 'costs.jsonl');
+    const row = JSON.parse(fs.readFileSync(metricsFile, 'utf8').trim());
+    assert.strictEqual(row.estimated_cost_usd, 18, 'Expected Sonnet 4.6 1M/1M to remain $18.00');
+
+    fs.rmSync(tmpHome, { recursive: true, force: true });
+  }) ? passed++ : failed++);
```
[the remark this answers is not in the document]
Comment (santhiprakash) [reply tests/hooks/cost-tracker.test.js:468]: Addressed in commit e4ebb31. tests/hooks/cost-tracker.test.js now covers Sonnet 5 cache write/read rates, a dated Sonnet 5 model ID, and the claude-sonnet-50 near-miss falling back to standard Sonnet rates. All 14 tests pass.

**Comment by santhiprakash** [comment]
CodeRabbit's test-isolation suggestion is addressed in 3569a0d — each Sonnet 5, cache, dated, and near-miss test now uses a unique session ID (suffixed with `Date.now()`) and removes its temp directory in cleanup. `node tests/hooks/cost-tracker.test.js` still passes 14/14. No further author action needed unless a maintainer requests changes.

**Comment by santhiprakash** [comment]
Addressed the cache-isolation finding in `baa40f7d`.

- Pricing tests now use process-scoped session IDs, remove any existing harness cache before execution, and remove it in `finally` cleanup.
- The Sonnet 5 test seeds a conflicting fresh cache; without isolation it records `999`, while the isolated test records the expected `$12` transcript estimate.
- `node tests/hooks/cost-tracker.test.js` → **14 passed, 0 failed**.
- `git diff --check` is clean.

The repository lint command could not start in this checkout because dependencies are not installed (`eslint` / `@eslint/js` missing).

**Comment by santhiprakash** [comment]
Addressed in a0c1bb46: added independent Sonnet 5 cache-write ($14.50) and cache-read ($12.20) regression cases while retaining the combined-rate check.

Verification: `node tests/hooks/cost-tracker.test.js` (16 passed) and `npx eslint scripts/hooks/cost-tracker.js tests/hooks/cost-tracker.test.js` (clean). `git diff --check` is clean.

**Comment by haelyra** [comment]
Thank you for catching the Sonnet 5 pricing mismatch and adding focused cache-path regression coverage. Your fix has now landed on main through #2866 as part of the connected cost-tracker integration, and the exact resulting main commit passed CI and CodeQL. We appreciate the precise correction and the tests that made it safe to ship.
